control-plane/agents/src/bin/core/controller/reconciler/nexus/mod.rs

```diff
@@ -130,61 +135,126 @@ async fn nexus_reconciler(
+        info!(%child.uri, "Child's Replica is not online yet");
     }
+    Ok(())
+}
 
-    PollResult::Ok(PollerState::Idle)
+/// Removes child from nexus children list to initiate Full rebuild of child.
+#[tracing::instrument(skip(nexus, context), level = "trace", fields(nexus.uuid = %nexus.uuid(), request.reconcile = true))]
+pub(super) async fn faulted_children_remover(
+    nexus: &mut OperationGuardArc<NexusSpec>,
+    child: &Child,
+    context: &PollContext,
+) -> Result<(), SvcError> {
+    let nexus_uuid = nexus.uuid();
+    let nexus_state = context.registry().nexus(nexus_uuid).await?;
+    let nexus_spec_clone = nexus.lock().clone();
+    nexus_spec_clone.warn_span(|| {
+        tracing::warn!(%child.uri, %child.state, %child.state_reason, "Attempting to remove faulted child")
+    });
+    nexus
+        .remove_child_by_uri(context.registry(), &nexus_state, &child.uri, true)
+        .await?;
+
+    nexus_spec_clone.warn_span(|| {
+        tracing::warn!(%child.uri, %child.state, %child.state_reason, "Successfully removed faulted child")
+    });
+    Ok(())
+}
+
+/// Returns true if time elapsed from child fault time is greater or equal to wait time duration.
+fn is_time_elapsed(
+    fault_time: Option<SystemTime>,
+    wait_duration: Duration,
+    uri: &ChildUri,
+) -> bool {
+    if let Some(fault_time) = fault_time {
+        if let Ok(elapsed) = fault_time.elapsed() {
+            info!(
+                child.uri = %uri,
+                "waited {:?} for child to comeback. Wait time is {:?}",
+                elapsed, wait_duration
+            );
+            return elapsed >= wait_duration;
+        }
+    }
+    false
+}
+
+/// If we can get replica for a child from registry. It means that child node is online, Pool is
+/// imported containing the child/replica.
+async fn get_child_replica(
+    nexus: &NexusSpec,
+    child: &Child,
+    context: &PollContext,
+) -> Result<Replica, SvcError> {
+    let replica_uri = nexus.replica_uri(&child.uri).ok_or(SvcError::NotFound {
+        kind: ResourceKind::Replica,
+        id: "Child replica not found".to_string(),
+    })?;
+    context.registry().replica(replica_uri.uuid()).await
+}
+
+/// Tries to mark child as Online. Returns error on failing to do so.
+pub(super) async fn initialize_partial_rebuild(
+    nexus: &mut OperationGuardArc<NexusSpec>,
+    child: &ChildUri,
+    context: &PollContext,
+) -> Result<(), SvcError> {
+    let nexus_node = &nexus.as_ref().node;
+    info!(%child, "Making online of {:?} nexus", nexus);
+    let node = context.registry().node_wrapper(nexus_node).await?;
+    let online_context = NexusChildActionContext::new(nexus_node, nexus.uuid(), child);
+    let ctx = NexusChildActionContextNode::new(online_context, context.registry());
+    // TODO: Check if we can handle things differently for different SvcError.
+    node.online_child(ctx).await?;
+    info!(
+        %child, "Successfully made child online in nexus :{:?}", nexus
```

Comment:
We don't need to trace the entire nexus, you can take a look at existing traces.

Reply:
Yes, used to be NexusId before. Made changes